APE13.rst

```diff
@@ -0,0 +1,409 @@
+``dec=`` set to the second axis, and the ``frame=fk5`` and ``equinox=J2005``
+arguments, while the second world axis can be used to instantiate an Astropy
+``Time`` object as the first positional argument, and with the ``scale=tai``
+keyword argument. Note that the coordinate frame classes could be custom
+sub-classes if needed.
+
+Low-level API examples
+^^^^^^^^^^^^^^^^^^^^^^
+
+**Simple 1D spectrum** - 1D mapping from pixel to wavelength:
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True]]
+    wcs.world_axis_units = [‘angstrom’]
+    wcs.world_axis_physical_type = [‘spectral’]
+    wcs.world_axis_object_components = [(‘spec’, 0)]
+    wcs.world_axis_object_classes  = {‘spec’:(‘astropy.units.Wavelength’:
+                                              {‘airorvacwl’: ‘air’})}
+
+**Simple 2D image mapping** where the axis are just RA and Dec (in WCS this would be CAR)
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True, False], [False, True]]
+    wcs.world_axis_units = [‘deg’, ‘deg’]
+    wcs.world_axis_physical_type = [‘pos.eq.ra’, ‘pos.ra.dec’]
+    wcs.world_axis_object_components = [((‘sc’, ‘ra’), (‘sc’, ‘dec’)]
+    wcs.world_object_data  = {‘sc’:(‘astropy.coordinates.SkyCoord’,
+                                {‘frame’: ‘icrs’})}
+
+**Extremely complex spectral data cube** with 3 *pixel* dimensions and 4 *world*
+dimensions. First two *pixel* dimensions encode a mixed set of spatial
+dimensions and a 3rd dimensions that’s completely spectral (i.e., the output of
+an IFU detector), and the third *pixel* dimension is a separable 4th world
+dimension encoding time-of-observation.
+
+.. code-block:: python
+
+    axis_correlation_matrix = [[True, True, False],
+                               [True, True, False],
+                               [True, True, False],
+                               [False, False, True]]
+    world_axis_units = [‘deg’, ‘deg’, ‘angstrom’, ‘day’]
+    world_axis_physical_type = [‘pos.galactic.lon’, ‘pos.galactic.lat’, ‘em.freq’, ‘time’]
+    world_axis_object_components = [(‘spat’, ‘ra’), (‘spat’, ‘dec’),
+                                    (‘spec’, 0), (‘time’, 0)]
+    world_object_data  = {‘spat’: (‘astropy.coordinates.SkyCoord’,
+                                   {‘frame’: ‘icrs’}),
+                          ‘spec’: (‘astropy.units.Wavelength`, {}),
+                          ‘time’: (‘astropy.time.Time’,
+                                   {‘format’:’mjd’, ‘scale’:’utc’})}
+
+**The identity transform** for a 1D array (i.e., pixel -> pixel):
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True]]
+    wcs.world_axis_units = [‘pixel’]
+    wcs.world_axis_physical_type = [‘instr.pixel’]
+    wcs.world_axis_object_components = [(‘spec’, 0)]
+    wcs.world_axis_object_classes  = {‘spec’:(‘astropy.units.pixel’: {})}
+
+Common UCD1+ names
+^^^^^^^^^^^^^^^^^^
+
+The full UCD1+ vocabulary (used for the ``world_physical_types`` attribute)
```

Comment:
How can you represent spatial coordinate systems not explicitly in the UCD1+ list? Like any of the solar coordinate systems and quite a few others.

Reply:
This is a valid question - if we decide to go ahead with UCD1+ then we can either just work with the IVOA to add other names as needed and just return None in the mean time, or we can come up with a list of additional acceptable attributes. While frustrating because things might be slow, I think the former would be better.

A complete alternative would be to use much more general terms such as celestial, spectral, etc. and just come up with the list ourselves (basically with terms that align with the STC standard)

@eteq @keflavich - any thoughts?

Reply:
@Cadair @eteq @keflavich - see 362b28d0f9dd6c269ac1ab45ce2002fed45f97e5

Reply:
I think that new change looks good.  But maybe you should add this to the list of "alternative", @astrofrog?  Also I have some minor word-smithing changes but can just do that myself.